File: src/main/python/net/coderextreme/data/BoxEm.py
print('<!--')
import x3d
print('-->')
X3D0 = x3d.X3D()
X3D0.profile = "Immersive"
X3D0.version = "4.0"
head1 = x3d.head()
meta2 = x3d.meta()
meta2.name = "title"
meta2.content = "BoxEm.x3d"

head1.children.append(meta2)
meta3 = x3d.meta()
meta3.name = "creator"
meta3.content = "John Carlson"

head1.children.append(meta3)
meta4 = x3d.meta()
meta4.name = "generator"
meta4.content = "manual"

head1.children.append(meta4)
meta5 = x3d.meta()
meta5.name = "identifier"
meta5.content = "https://coderextreme.net/X3DJSONLD/src/main/data/box.x3d"

head1.children.append(meta5)
meta6 = x3d.meta()
meta6.name = "description"
meta6.content = "3 boxes"

head1.children.append(meta6)

X3D0.head = head1
Scene7 = x3d.Scene()
NavigationInfo8 = x3d.NavigationInfo()
NavigationInfo8.type = ["EXAMINE"]

Scene7.children.append(NavigationInfo8)
Viewpoint9 = x3d.Viewpoint()
Viewpoint9.description = "Cubes on Fire"
Viewpoint9.position = [0,0,12]

Scene7.children.append(Viewpoint9)
ProtoDeclare10 = x3d.ProtoDeclare()
ProtoDeclare10.name = "anyShape"
ProtoInterface11 = x3d.ProtoInterface()
field12 = x3d.field()
field12.name = "xtranslation"
field12.accessType = "inputOutput"
field12.type = "SFVec3f"
field12.value = [0,0,0]

ProtoInterface11.field.append(field12)
field13 = x3d.field()
field13.name = "myShape"
field13.accessType = "inputOutput"
field13.type = "MFNode"
Shape14 = x3d.Shape()
Sphere15 = x3d.Sphere()

Shape14.geometry = Sphere15
Appearance16 = x3d.Appearance()
Material17 = x3d.Material()
Material17.diffuseColor = [1,1,1]

Appearance16.material = Material17

Shape14.appearance = Appearance16

field13.children.append(Shape14)

ProtoInterface11.field.append(field13)

ProtoDeclare10.ProtoInterface = ProtoInterface11
ProtoBody18 = x3d.ProtoBody()
Transform19 = x3d.Transform()
IS20 = x3d.IS()
connect21 = x3d.connect()
connect21.nodeField = "translation"
connect21.protoField = "xtranslation"

IS20.connect.append(connect21)
connect22 = x3d.connect()
connect22.nodeField = "children"
connect22.protoField = "myShape"

IS20.connect.append(connect22)

Transform19.IS = IS20

ProtoBody18.children.append(Transform19)

ProtoDeclare10.ProtoBody = ProtoBody18

Scene7.children.append(ProtoDeclare10)
ProtoDeclare23 = x3d.ProtoDeclare()
ProtoDeclare23.name = "three"
ProtoInterface24 = x3d.ProtoInterface()
field25 = x3d.field()
field25.name = "ytranslation"
field25.accessType = "inputOutput"
field25.type = "SFVec3f"
field25.value = [0,0,0]

ProtoInterface24.field.append(field25)
field26 = x3d.field()
field26.name = "myShape"
field26.accessType = "inputOutput"
field26.type = "MFNode"
Shape27 = x3d.Shape()
Cylinder28 = x3d.Cylinder()

Shape27.geometry = Cylinder28
Appearance29 = x3d.Appearance()
Material30 = x3d.Material()
Material30.diffuseColor = [1,1,1]

Appearance29.material = Material30

Shape27.appearance = Appearance29

field26.children.append(Shape27)

ProtoInterface24.field.append(field26)

ProtoDeclare23.ProtoInterface = ProtoInterface24
ProtoBody31 = x3d.ProtoBody()
Transform32 = x3d.Transform()
IS33 = x3d.IS()
connect34 = x3d.connect()
connect34.nodeField = "translation"
connect34.protoField = "ytranslation"

IS33.connect.append(connect34)

Transform32.IS = IS33
ProtoInstance35 = x3d.ProtoInstance()
ProtoInstance35.name = "anyShape"
fieldValue36 = x3d.fieldValue()
fieldValue36.name = "xtranslation"
fieldValue36.value = "0 0 0"

ProtoInstance35.fieldValue.append(fieldValue36)
IS37 = x3d.IS()
connect38 = x3d.connect()
connect38.nodeField = "myShape"
connect38.protoField = "myShape"

IS37.connect.append(connect38)

ProtoInstance35.IS = IS37

Transform32.children.append(ProtoInstance35)
ProtoInstance39 = x3d.ProtoInstance()
ProtoInstance39.name = "anyShape"
fieldValue40 = x3d.fieldValue()
fieldValue40.name = "xtranslation"
fieldValue40.value = "2 0 0"

ProtoInstance39.fieldValue.append(fieldValue40)
IS41 = x3d.IS()
connect42 = x3d.connect()
connect42.nodeField = "myShape"
connect42.protoField = "myShape"

IS41.connect.append(connect42)

ProtoInstance39.IS = IS41

Transform32.children.append(ProtoInstance39)
ProtoInstance43 = x3d.ProtoInstance()
ProtoInstance43.name = "anyShape"
fieldValue44 = x3d.fieldValue()
fieldValue44.name = "xtranslation"
fieldValue44.value = "-2 0 0"

ProtoInstance43.fieldValue.append(fieldValue44)
IS45 = x3d.IS()
connect46 = x3d.connect()
connect46.nodeField = "myShape"
connect46.protoField = "myShape"

IS45.connect.append(connect46)

ProtoInstance43.IS = IS45

Transform32.children.append(ProtoInstance43)

ProtoBody31.children.append(Transform32)

ProtoDeclare23.ProtoBody = ProtoBody31

Scene7.children.append(ProtoDeclare23)
ProtoInstance47 = x3d.ProtoInstance()
ProtoInstance47.name = "three"
ProtoInstance47.DEF = "threepi"
fieldValue48 = x3d.fieldValue()
fieldValue48.name = "ytranslation"
fieldValue48.value = "0 0 0"

ProtoInstance47.fieldValue.append(fieldValue48)
fieldValue49 = x3d.fieldValue()
fieldValue49.name = "myShape"
Shape50 = x3d.Shape()
Shape50.DEF = "box"
Box51 = x3d.Box()
Box51.size = [1,1,1]

Shape50.geometry = Box51
Appearance52 = x3d.Appearance()
Material53 = x3d.Material()
Material53.diffuseColor = [0,1,0]

Appearance52.material = Material53

Shape50.appearance = Appearance52

fieldValue49.children.append(Shape50)

ProtoInstance47.fieldValue.append(fieldValue49)

Scene7.children.append(ProtoInstance47)
Transform54 = x3d.Transform()
Transform54.translation = [0,2,0]
Shape55 = x3d.Shape()
Shape55.USE = "box"

Transform54.children.append(Shape55)

Scene7.children.append(Transform54)

X3D0.Scene = Scene7
f = open("../data/BoxEm.new.python.x3d", mode="w", encoding="utf-8")
f.write(X3D0.XML())
f.close()
f = open("../data/BoxEm.new.python.x3dv", mode="w", encoding="utf-8")
f.write(X3D0.VRML())
f.close()
f = open("../data/BoxEm.new.python.x3dj", mode="w", encoding="utf-8")
f.write(X3D0.JSON())
f.close()

File: src/main/java/net/coderextreme/data/BoxEm.java
package net.coderextreme.data;
import org.web3d.x3d.jsail.*;
import org.web3d.x3d.jsail.CADGeometry.*;
import org.web3d.x3d.jsail.Core.*;
import org.web3d.x3d.jsail.CubeMapTexturing.*;
import org.web3d.x3d.jsail.DIS.*;
import org.web3d.x3d.jsail.EnvironmentalEffects.*;
import org.web3d.x3d.jsail.EnvironmentalSensor.*;
import org.web3d.x3d.jsail.EventUtilities.*;
import org.web3d.x3d.jsail.Followers.*;
import org.web3d.x3d.jsail.Geometry2D.*;
import org.web3d.x3d.jsail.Geometry3D.*;
import org.web3d.x3d.jsail.Geospatial.*;
import org.web3d.x3d.jsail.Grouping.*;
import org.web3d.x3d.jsail.HAnim.*;
import org.web3d.x3d.jsail.Interpolation.*;
import org.web3d.x3d.jsail.KeyDeviceSensor.*;
import org.web3d.x3d.jsail.Layering.*;
import org.web3d.x3d.jsail.Layout.*;
import org.web3d.x3d.jsail.Lighting.*;
import org.web3d.x3d.jsail.NURBS.*;
import org.web3d.x3d.jsail.Navigation.*;
import org.web3d.x3d.jsail.Networking.*;
import org.web3d.x3d.jsail.ParticleSystems.*;
import org.web3d.x3d.jsail.Picking.*;
import org.web3d.x3d.jsail.PointingDeviceSensor.*;
import org.web3d.x3d.jsail.Rendering.*;
import org.web3d.x3d.jsail.RigidBodyPhysics.*;
import org.web3d.x3d.jsail.Scripting.*;
import org.web3d.x3d.jsail.Shaders.*;
import org.web3d.x3d.jsail.Shape.*;
import org.web3d.x3d.jsail.Sound.*;
import org.web3d.x3d.jsail.Text.*;
import org.web3d.x3d.jsail.Texturing3D.*;
import org.web3d.x3d.jsail.Texturing.*;
import org.web3d.x3d.jsail.Time.*;
import org.web3d.x3d.jsail.VolumeRendering.*;
import org.web3d.x3d.jsail.fields.*;
import java.util.ArrayList;
import java.util.List;
import net.coderextreme.X3DRoots;
public class BoxEm implements X3DRoots {
  public static void main(String[] args) {
    ConfigurationProperties.setXsltEngine(ConfigurationProperties.XSLT_ENGINE_NATIVE_JAVA);
    ConfigurationProperties.setDeleteIntermediateFiles(false);
    ConfigurationProperties.setStripTrailingZeroes(true);
    ConfigurationProperties.setStripDefaultAttributes(true);
    X3D model = new BoxEm().getRootNodeList().get(0); // only get one root node
    System.out.print(model.validationReport().trim());
    model.toFileX3D("../data/BoxEm.new.java.x3d");
    model.toFileJSON("../data/BoxEm.new.java.json");
    }
    public List<X3D> getRootNodeList() {
    	List<X3D> list = new ArrayList<X3D>(1);
    	list.add(initialize());
    	return list;
    }
    public X3D initialize() {
ProtoInstance ProtoInstance0 = null;
ProtoInstance ProtoInstance1 = null;
ProtoInstance ProtoInstance2 = null;
ProtoInstance ProtoInstance3 = null;
      X3D X3D0 =  new X3D().setProfile("Immersive").setVersion("4.0")
      .setHead(new head()
        .addMeta(new meta().setName("title").setContent("BoxEm.x3d"))
        .addMeta(new meta().setName("creator").setContent("John Carlson"))
        .addMeta(new meta().setName("generator").setContent("manual"))
        .addMeta(new meta().setName("identifier").setContent("https://coderextreme.net/X3DJSONLD/src/main/data/box.x3d"))
        .addMeta(new meta().setName("description").setContent("3 boxes")))
      .setScene(new Scene()
        .addChild(new NavigationInfo().setType("\"EXAMINE\""))
        .addChild(new Viewpoint().setDescription("Cubes on Fire").setPosition(new double[] {0,0,12}))
        .addChild(new ProtoDeclare().setName("anyShape")
          .setProtoInterface(new ProtoInterface()
            .addField(new field().setType("SFVec3f").setName("xtranslation").setAccessType(field.ACCESSTYPE_INPUTOUTPUT).setValue("0 0 0"))
            .addField(new field().setType("MFNode").setName("myShape").setAccessType(field.ACCESSTYPE_INPUTOUTPUT)
              .addChild(new Shape()
                .setGeometry(new Sphere())
                .setAppearance(new Appearance()
                  .setMaterial(new Material().setDiffuseColor(new double[] {1,1,1}))))))
          .setProtoBody(new ProtoBody()
            .addChild(new Transform()
              .setIS(new IS()
                .addConnect(new connect().setNodeField("translation").setProtoField("xtranslation"))
                .addConnect(new connect().setNodeField("children").setProtoField("myShape"))))))
        .addChild(new ProtoDeclare().setName("three")
          .setProtoInterface(new ProtoInterface()
            .addField(new field().setType("SFVec3f").setName("ytranslation").setAccessType(field.ACCESSTYPE_INPUTOUTPUT).setValue("0 0 0"))
            .addField(new field().setType("MFNode").setName("myShape").setAccessType(field.ACCESSTYPE_INPUTOUTPUT)
              .addChild(new Shape()
                .setGeometry(new Cylinder())
                .setAppearance(new Appearance()
                  .setMaterial(new Material().setDiffuseColor(new double[] {1,1,1}))))))
          .setProtoBody(new ProtoBody()
            .addChild(new Transform()
              .setIS(new IS()
                .addConnect(new connect().setNodeField("translation").setProtoField("ytranslation")))
              .addChild(ProtoInstance0 = new ProtoInstance().setName("anyShape")
                .setIS(new IS()
                  .addConnect(new connect().setNodeField("myShape").setProtoField("myShape"))))
              .addChild(ProtoInstance1 = new ProtoInstance().setName("anyShape")
                .setIS(new IS()
                  .addConnect(new connect().setNodeField("myShape").setProtoField("myShape"))))
              .addChild(ProtoInstance2 = new ProtoInstance().setName("anyShape")
                .setIS(new IS()
                  .addConnect(new connect().setNodeField("myShape").setProtoField("myShape")))))))
        .addChild(ProtoInstance3 = new ProtoInstance().setName("three").setDEF("threepi"))
        .addChild(new Transform().setTranslation(new double[] {0,2,0})
          .addChild(new Shape().setUSE("box"))))      ;
ProtoInstance0
                .addFieldValue(new fieldValue().setName("xtranslation").setValue("0 0 0"));
ProtoInstance1
                .addFieldValue(new fieldValue().setName("xtranslation").setValue("2 0 0"));
ProtoInstance2
                .addFieldValue(new fieldValue().setName("xtranslation").setValue("-2 0 0"));
ProtoInstance3
          .addFieldValue(new fieldValue().setName("ytranslation").setValue("0 0 0"));
ProtoInstance3
          .addFieldValue(new fieldValue().setName("myShape")
            .addChild(new Shape().setDEF("box")
              .setGeometry(new Box().setSize(new double[] {1,1,1}))
              .setAppearance(new Appearance()
                .setMaterial(new Material().setDiffuseColor(new double[] {0,1,0})))));
    return X3D0;
    }
}
